### ISSUE TITLE
Paper #422

### ISSUE THREAD

**Issue by kstolee** [researchart/fse20#102]
This issue is for discussion related to PR #26 

Author contact: @kripa432

**Comment by kstolee** [comment]
## Authors

Kripa Shanker (corresponding author)  
kripashanker@iisc.ac.in  
Github: [kripa432](https://github.com/kripa432)  

Arun Joseph  
arunj@iisc.ac.in  

Vinod Ganapathy  
vg@iisc.ac.in

**Comment by art-reviewer** [comment]
[Paper](https://github.com/kripa432/fse20/blob/ab2346b852dfeb47979a64c20c0149472b35bede/submissions/reusable/sgx-porting-methods/paper.pdf)

**Comment by anonReviewer10000** [comment]
This artifact seems to meet the requirements for a reusable badge:

Documented: This artifact contains documentation showing exactly which commands to run to obtain answers to each RQ in the paper.  The version is easily known via the Github repository.  The requirements are also easily found.  More details could be provided about exactly what each command does, though this information can mostly be assumed from the code and/or the FSE paper.

Consistent: The artifact materials are clearly in line with the FSE paper.  Each component is associated with one RQ.

Complete: I was able to obtain all software dependencies.  However, I could not run the tools because I do not have a Skylake CPU.

Exercisable: As mentioned, all details are associated with RQs in the paper.  There seems to be a clear path from provided scripts to results in the paper.  However, I was not able to verify due to hardware requirements.

**Comment by anon-eval** [comment]
@kripa432 
Thank you for submitting the artifacts. Congratulations on the paper :)

**TLDR**: Based on the current version of your submission, I cannot recommend the **reusable** badge(yet) although the framework is `functional` and `available`. Please update the `ReadMe` on how a  the evaluation framework can be expanded to support new tool. 

I arrived at this conclusion using the following metrics
1. **Installation** ✅ : The installation runs without any major hiccups when the instructions in `Install.md` is followed.
2. **Public Availability**✅ : The tool is publicly available on github. I would recommend the authors to upload the tool on [doi.org](doi.org) or [zenodo](zenodo.org) as well.
3. **Documentation** ✅  : The tool is well documented. The authors also do a great job in providing alterate scripts for installation.
4. **Consistent/Complete**✅ : The results are consistent on multiple executions and the results in the paper can be reproduced based on the documentation
5. **Exercisable** ✅ : The evaluation framework is exercisable on the tools provided in the repository
6. **Evidence of verification and validation** ✅  : I was able to only validate a the results for the research questions using the scripts provided.
7 **Reusability** ❌ : The authors should mention how a new tool can be run for new projects not mentioned in this study.

Thanks :)

**Comment by kripa432** [comment]
Dear reviewers, 

  we want to thank you for your valuable review #102 . 
As per the review of @anon-eval we have done the following:-

1. For available badge, we have archived Porpoise at [https://doi.org/10.5281/zenodo.3895761](https://doi.org/10.5281/zenodo.3895761)
2. We have added a tutorial in README.md file to port libjpeg to Intel SGX with Porpoise. Also we have added a screencast of same tutorial.

I have created a new pull request #131 

Thanks
Authors

**Comment by anon-eval** [comment]
@kripa432
Thank you for making the update. The screencast should help other users to expand the evaluation framework for more tools. Based on your changes I would recommend the `reusable` badge.

**Comment by kstolee** [comment]
Great, thanks everyone! @anon-eval and @anonReviewer10000 you seem to be in agreement that the ```reusable``` and ```available``` badges are appropriate. I'll add the labels. If you have further comments or concerns, please keep the conversation going.

**Comment by kripa432** [comment]
Dear @kstolee 

Based on the prior reviews of reviewers, should we also get `functional` badge or we are missing something?

Thanks
Authors

**Comment by kstolee** [comment]
@kripa432 You are correct; this was *my* misunderstanding. You can (and you have!) earn three badges.
